bootloader/wscript

```diff
@@ -188,43 +192,64 @@ def set_lsb_compiler(ctx):
     ctx.env.append_value('LINKFLAGS', lsb_target_flag)
 
 
+def check_sizeof_pointer(ctx):
+
+    def check(type, expected):
+        # test code taken from autoconf resp. Scons: this is a pretty clever
+        # hack to find that a type is of a given size using only compilation.
+        # This speeds things up quite a bit compared to straightforward code
+        # actually running the code.
+        # Plus: This works cross :-)
+        fragment = '''
+        int main() {
+            static int test_array[1 - 2 * !(((long int) (sizeof(%s))) == %d)];
+            test_array[0] = 0;
+            return 0;
+        }''' % (type, expected)
+        return ctx.check_cc(fragment=fragment, execute=False, mandatory=False)
+
```

Comment:
@htgoebel `sizeof` returns an unsigned type, which should not be cast to a `long int`, which is signed.